extensions/amp-form/0.1/amp-form.js

```diff
@@ -431,6 +432,15 @@ export class AmpForm {
       checkUserValidityAfterInteraction_(dev().assertElement(e.target));
       this.validator_.onInput(e);
     });
+
+    // Ctrl/Cmd + Enter on textarea or contenteditable triggers form submission.
+    this.form_.querySelectorAll('textarea,[contenteditable]').forEach(el =>
```

Comment:
Why not a single event listener on the form itself and check the event target?

Reply:
That's fine too. Let me see if I understand the rationale: Registering on all relevant selectors has a guaranteed upfront cost whereas registering on the form reduces the work of checking to only when form submission is likely.

Reply:
I believe there's some perf overhead for event listeners itself. I'd actually consider having a single document-level listener in `AmpFormService`. This would also handle the case where inputs are not descendants of their target form.

Reply:
How do inputs specify a target form that they are not a descendent of?

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefform

> The form attribute lets you place an input anywhere in the document but have it included with a form elsewhere in the document.

Reply:
Note that amp-form's current support for non-descendant inputs is highly broken in many places. :)

Reply:
Gave this a try on `AmpFormService` and removed `[contenteditable]` in the process since it isn't associated with the original bug.